Cannot find macro 'llvm_asm' in this scope

I am following the guide [here](https://espressif-trainings.ferrous-systems.com/02_4_hello_board.html) and when trying to run the _hardware-check_ example:
`cargo espflash --release --monitor /dev/tty.usbserial-1130`

I get the following error:
`error: could not compile "riscv" due to 366 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
error: cannot find macro "llvm_asm" in this scope`

`error[E0635]: unknown feature "llvm_asm"
  --> /Users/reinhard/.cargo/registry/src/github.com-1ecc6299db9ec823/riscv-0.7.0/src/lib.rs:17:45
   |
17 | #![cfg_attr(feature = "inline-asm", feature(llvm_asm))]`

I have tried pinning nightly to nightly-2022-03-10 (using `rustup install nightly-2022-03-10` and `rustup component add rust-src --toolchain nightly-2022-03-10`) and to 2022-04-21, but both result in errors.

--- a/advanced/button-interrupt/exercise/Cargo.toml
+++ b/advanced/button-interrupt/exercise/Cargo.toml
@@ -26,4 +26,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/advanced/button-interrupt/solution/Cargo.toml
+++ b/advanced/button-interrupt/solution/Cargo.toml
@@ -26,4 +26,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/advanced/i2c-driver/solution/Cargo.toml
+++ b/advanced/i2c-driver/solution/Cargo.toml
@@ -28,4 +28,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/advanced/i2c-sensor-reading/solution/Cargo.toml
+++ b/advanced/i2c-sensor-reading/solution/Cargo.toml
@@ -33,4 +33,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/common/lib/esp32-c3-dkc02-bsc/Cargo.toml
+++ b/common/lib/esp32-c3-dkc02-bsc/Cargo.toml
@@ -19,4 +19,4 @@ toml-cfg = "0.1"
 riscv = { version = "0.7", features=["inline-asm"] }
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/intro/hardware-check/Cargo.toml
+++ b/intro/hardware-check/Cargo.toml
@@ -28,4 +28,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/intro/http-client/exercise/Cargo.toml
+++ b/intro/http-client/exercise/Cargo.toml
@@ -29,4 +29,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/intro/http-client/solution/Cargo.toml
+++ b/intro/http-client/solution/Cargo.toml
@@ -29,4 +29,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/intro/http-server/exercise/Cargo.toml
+++ b/intro/http-server/exercise/Cargo.toml
@@ -30,4 +30,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/intro/http-server/solution/Cargo.toml
+++ b/intro/http-server/solution/Cargo.toml
@@ -30,4 +30,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/intro/mqtt/exercise/Cargo.toml
+++ b/intro/mqtt/exercise/Cargo.toml
@@ -35,4 +35,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

--- a/intro/mqtt/solution/Cargo.toml
+++ b/intro/mqtt/solution/Cargo.toml
@@ -35,4 +35,4 @@ embuild = "0.28"
 anyhow = "1"
 
 [patch.crates-io]
-riscv = { git = "https://github.com/rust-embedded/riscv", branch = "master"}
+riscv = { git = "https://github.com/rust-embedded/riscv", rev = "396fb9b"}

